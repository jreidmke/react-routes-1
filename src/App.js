import logo from './logo.svg';
import './App.css';
import { Route, Redirect } from "react-router-dom";
import DogList from './DogList';
import DogDetails from './DogDetails';

import whiskey from "./images/whiskey.jpg";
import tubby from "./images/tubby.jpg";
import duke from "./images/duke.jpg";
import perry from "./images/perry.jpg";

function App() {
  return (
    <div className="App">
      <h1>Look at these Dogs!</h1>
        <Route exact path="/dogs">
          {dogs.dogs.map(d => <DogList name={d.name} age={d.age} src={d.src} facts={d.facts}/>)}
        </Route>

        <Route exact path="/dogs/:name">
          <DogDetails dogs={dogs.dogs}/>
        </Route>

        <Redirect to="/dogs"/>
    </div>
  );
}

export default App;

export const dogs = {
  dogs: [
    {
      name: "Whiskey",
      age: 5,
      src: whiskey,
      facts: [
        "Whiskey loves eating popcorn.",
        "Whiskey is a terrible guard dog.",
        "Whiskey wants to cuddle with you!"
      ]
    },
    {
      name: "Duke",
      age: 3,
      src: duke,
      facts: [
        "Duke believes that ball is life.",
        "Duke likes snow.",
        "Duke enjoys pawing other dogs."
      ]
    },
    {
      name: "Perry",
      age: 4,
      src: perry,
      facts: [
        "Perry loves all humans.",
        "Perry demolishes all snacks.",
        "Perry hates the rain."
      ]
    },
    {
      name: "Tubby",
      age: 4,
      src: tubby,
      facts: [
        "Tubby is really stupid.",
        "Tubby does not like walks.",
        "Angelina used to hate Tubby, but claims not to anymore."
      ]
    }
  ]
}


App.defaultProps = { dogs };