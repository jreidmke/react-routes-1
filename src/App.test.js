import { render, fireEvent } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';

test('renders app', () => {
  const {getByText} = render(
    <MemoryRouter initialEntries={['/dogs']}>
      <App/>
    </MemoryRouter>
  );
  expect(getByText('Duke enjoys pawing other dogs.')).toBeInTheDocument();
})

test('matches snapshot', () => {
  const {asFragment} = render(
    <MemoryRouter initialEntries={['/dogs']}>
      <App/>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
})

test('links take user to dog details page', () => {
  const {getByText} =render(
    <MemoryRouter initialEntries={['/dogs']}>
      <App/>
    </MemoryRouter>
  );
  expect(getByText("Look at these Dogs!")).toBeInTheDocument();

  fireEvent.click(getByText('Whiskey'));

  expect(getByText('Return Home')).toBeInTheDocument();

  fireEvent.click(getByText('Return Home'));

  expect(getByText("Look at these Dogs!")).toBeInTheDocument();
})