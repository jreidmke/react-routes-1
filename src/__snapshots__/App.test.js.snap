// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Look at these Dogs!
    </h1>
    <div>
      <img
        src="whiskey.jpg"
      />
      <h3>
        <a
          href="/dogs/whiskey"
        >
          Whiskey
        </a>
        , 5 Years Old
      </h3>
      <ul>
        <li>
          Whiskey loves eating popcorn.
        </li>
        <li>
          Whiskey is a terrible guard dog.
        </li>
        <li>
          Whiskey wants to cuddle with you!
        </li>
      </ul>
    </div>
    <div>
      <img
        src="duke.jpg"
      />
      <h3>
        <a
          href="/dogs/duke"
        >
          Duke
        </a>
        , 3 Years Old
      </h3>
      <ul>
        <li>
          Duke believes that ball is life.
        </li>
        <li>
          Duke likes snow.
        </li>
        <li>
          Duke enjoys pawing other dogs.
        </li>
      </ul>
    </div>
    <div>
      <img
        src="perry.jpg"
      />
      <h3>
        <a
          href="/dogs/perry"
        >
          Perry
        </a>
        , 4 Years Old
      </h3>
      <ul>
        <li>
          Perry loves all humans.
        </li>
        <li>
          Perry demolishes all snacks.
        </li>
        <li>
          Perry hates the rain.
        </li>
      </ul>
    </div>
    <div>
      <img
        src="tubby.jpg"
      />
      <h3>
        <a
          href="/dogs/tubby"
        >
          Tubby
        </a>
        , 4 Years Old
      </h3>
      <ul>
        <li>
          Tubby is really stupid.
        </li>
        <li>
          Tubby does not like walks.
        </li>
        <li>
          Angelina used to hate Tubby, but claims not to anymore.
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
